File: 2540.最小公共值.go
/*
 * @lc app=leetcode.cn id=2540 lang=golang
 *
 * [2540] 最小公共值
 */

// @lc code=start
func getCommon(nums1 []int, nums2 []int) int {
	sort.Ints(nums1)
	sort.Ints(nums2)
	i, j := 0, 0
	for i < len(nums1) && j < len(nums2) {
		if nums1[i] == nums2[j] {
			return nums1[i]
		} else if nums1[i] < nums2[j] {
			i++
		} else {
			j++
		}
	}
	return -1
}

// @lc code=end

File: 2540.最小公共值.py
#
# @lc app=leetcode.cn id=2540 lang=python3
#
# [2540] 最小公共值
#
from typing import List
# @lc code=start
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        nums1.sort()
        nums2.sort()
        i = 0
        j = 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] == nums2[j]:
                return nums1[i]
            elif nums1[i] < nums2[j]:
                i += 1
            else:
                j += 1
        return -1
# @lc code=end

